galley/pkg/settings/conversion.go

```diff
@@ -0,0 +1,144 @@
+//  Copyright 2019 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package settings
+
+import (
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+
+	"istio.io/istio/galley/pkg/crd/validation"
+	"istio.io/istio/galley/pkg/server"
+	"istio.io/istio/pkg/ctrlz"
+	"istio.io/istio/pkg/mcp/creds"
+	"istio.io/istio/pkg/probe"
+)
+
+// ToProbeOptions converts the given Probe stanza to Probe options.
+func ToProbeOptions(p *Probe) (*probe.Options, error) {
+	if p.Disable {
+		return &probe.Options{}, nil
+	}
+
+	var interval time.Duration
+	if p.Interval != nil {
+		var err error
+		interval, err = ptypes.Duration(p.Interval)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &probe.Options{
+		Path:           p.Path,
+		UpdateInterval: interval,
+	}, nil
+}
+
+// ToServerArgs converts the given Galley settings to server.Args.
+func ToServerArgs(s *Galley) (*server.Args, error) {
+	var err error
+	var resyncPeriod time.Duration
+	var enableServer bool
+	var configPath string
+	var insecure bool
+	var accessListFile string
+	credentialOptions := creds.DefaultOptions()
+
+	k := s.Processing.Source.GetKubernetes()
+	if k != nil && k.ResyncPeriod != nil {
+		if resyncPeriod, err = ptypes.Duration(k.ResyncPeriod); err != nil {
```

Comment:
Shouldn't this use the `types.DurationFromProto` from `github.com/gogo/protobuf/types` if settings.proto is compiled with gogo proto?